fdbrpc/include/fdbrpc/ReplicationPolicy.h

```diff
@@ -73,6 +92,12 @@ struct IReplicationPolicy : public ReferenceCounted<IReplicationPolicy> {
 		return keys;
 	}
 	virtual void attributeKeys(std::set<std::string>*) const = 0;
+
+	mutable int _maxdepth; // -1 means uninitialized
```

Comment:
I assume we are caching these because we don't want to do the tree traversal every time "isSingleAcrossOverPolicyOne()" is called? Doesn't this variable (and "_depth") ever need to be cleared (similar to how it is done in "clearCache()")?

Reply:
Also, I assume we are declaring this as "mutable" because we are changing this from a const member function? I think a comment needs to be added above that const member function that it may update this variable, thanks!

Reply:
> It needs to be cleared when there is a configuration change, which calls clearCache().

But does "clearCache()" clear "_depth" (and "maxDepth")? It doesn't appear to be doing that (I could be wrong here, I don't fully understand this code).

Reply:
Yes, the locality structure will change when storage servers are added or removed and there is no reboot at that time. 

However, the replication policy does _not_ change in such cases. Adding or removing SS does not change if we are `tripple` or `double` so the depth and `maxdepth` should remain the same; the replication policy can only be changed via a configuration change which will reboot and create a branch new locality structure.

That said, it is difficult for me to prove there exist no edge cases (e.g. suppose the last SS in a zone is removed or last SS in a tripple configuration is lost). I will add this defensive check as it will be recomputed on the next call.